### PR TITLE
Update examples to show how to deal with extra validation copies

### PR DESCRIPTION
**Pull request by muellerzr**
# Update examples to show how to truncate the validation set for metrics

## What does this add?

Based off [this issue](https://github.com/huggingface/accelerate/issues/287) this PR updates all examples to show how to get rid of extra samples that get added when performing distributed training on the validation set.

Testing on a multigpu system will happen tommorow, but @sgugger pretty sure the way I have it setup ensures that this only runs when we have distributed systems, and that's where this problem arises?

## Who is it for?

Should close https://github.com/huggingface/accelerate/issues/287

## Why is it needed?

It's unclear from the scripts how to alleviate this behavior, and it's not documented anywhere. So, with this PR it now is

### CODE REVIEW

**Comment by muellerzr** [comment]
Note: This is just an initial to make sure the format and whatnot looks right and then all the other examples will follow suite :)

**Review by sgugger — COMMENTED** [review]
Thanks for the PR! I'd just put this in a specific feature example instead of the base one.

File: examples/by_feature/multi_node_metrics.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
+# coding=utf-8
```
Comment (sgugger) [inline examples/by_feature/multi_node_metrics.py:1]: Should be named `multi_process_metrics` as the problem arises as soon as you have several processes, not several nodes.

File: examples/by_feature/multi_node_metrics.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+# Copyright 2021 The HuggingFace Inc. team. All rights reserved.
```
Comment (sgugger) [inline examples/by_feature/multi_node_metrics.py:2]:
```suggestion
# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
```

**Review by sgugger — APPROVED** [review]
Thanks! Not sure what your question is about tests. To test this we would need to know in advance the exact value of the metric and make sure we get that again but it's very finnicky since the metric computed in the base script is roughly the same.

**Pull request merged**